### DISCUSSION TITLE
How to get the node features for link prediction task?

### DISCUSSION THREAD

**Discussion by shahinghasemi** [pyg-team/pytorch_geometric#5439] [General]
Hello, I have a heterogeneous network this is the data I have: 
types of nodes: `{dr, di, t, p, s, e}`
types of edges: `dr-p`, `dr-e`, `dr-t`, `dr-s`, `dr-di` which are undirected.
The final problem is to find the links in `dr-di`.

I extended the [provided](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py) example with my data. I have a big problem. I only have feature of node `di` and don't have any features for other nodes. for now I used identity matrix for other nodes types which is of shape `(N-node, N-node)` which `node` is our node types.

This is my question: How can I get the (initial)features in a reasonable way for other node types to use them for message-passing process to get enriched embeddings for my node types? Is using identity matrix for node features makes sense?

**Comment by rusty1s** [comment]
Yes, this is the usual appraoch. You can also use a learnable `torch.nn.Embedding` for each node type (which is similar but saves memory). Alternatively, there exists some work that explore using node embeddings generated by a `MetaPath2Vec` model.

**Reply by shahinghasemi** [reply]
Thank you for your reply. How about `edge-weight`? What's the usual way to get the edge-weight? Can I extract logical edge-weight with the adjacency matrices that I have?

**Reply by rusty1s** [reply]
If your adjacency matrix is weighted, the `edge_weight` should be given to you, right? Otherwise, there is no need to pass in an `edge_weight` vector in the first place. Let me know if I am missing something.
